### ISSUE TITLE
Multiplication of circulant matrices incorrect

### ISSUE THREAD

**Issue by daanhb** [JuliaMatrices/ToeplitzMatrices.jl#47]
The multiplication of circulant matrices does not give correct output:
```
julia> using ToeplitzMatrices

julia> C = Circulant(rand(4))
4×4 Circulant{Float64,Complex{Float64}}:
 0.811981  0.357227  0.461171  0.12133 
 0.12133   0.811981  0.357227  0.461171
 0.461171  0.12133   0.811981  0.357227
 0.357227  0.461171  0.12133   0.811981

julia> inv(C)*C
4×4 Circulant{Complex{Float64},Complex{Float64}}:
 0.495827+0.0778434im  0.713942-0.115764im   0.556134-0.0778434im  0.754495+0.115764im 
 0.754495+0.115764im   0.495827+0.0778434im  0.713942-0.115764im   0.556134-0.0778434im
 0.556134-0.0778434im  0.754495+0.115764im   0.495827+0.0778434im  0.713942-0.115764im 
 0.713942-0.115764im   0.556134-0.0778434im  0.754495+0.115764im   0.495827+0.0778434im

julia> collect(inv(C))*collect(C)
4×4 Array{Float64,2}:
 1.0          2.77556e-17   1.59595e-16  5.55112e-17
 5.55112e-17  1.0           1.38778e-17  1.11022e-16
 1.66533e-16  0.0           1.0          1.11022e-16
 0.0          1.11022e-16  -2.77556e-17  1.0  
```
The inverse circulant matrix is correct, but its internal representation is inconsistent and therefore subsequent computations fail. I found the error and I prepared a fix.

**Comment by dlfivefifty** [comment]
I'll merge a PR

**Comment by daanhb** [comment]
This is closed by #48.
